src/test/java/edu/ucla/library/bucketeer/verticles/ClearCacheIT.java

```diff
@@ -0,0 +1,372 @@
+                return putObjectS3(TIFFHORI_PATH);
+            }).compose(success -> {
+                return getInfoJsonObject();
+            });
+
+            return horiImageWidth.compose(horiWidth -> {
+                if (horiWidth != width) {
+                    return sendMessage(IMAGE_KEY);
+                }
+
+                return sendMessage(IMAGE_KEY).compose(success -> {
+                    return Future.failedFuture(LOGGER.getMessage(MessageCodes.BUCKETEER_610));
+                });
+            });
+        }).onSuccess(result -> {
+            myAmazonS3.deleteObject(s3Bucket, IMAGE_JPX);
+            TestUtils.complete(asyncTask);
+        }).onFailure(failure -> {
+            myAmazonS3.deleteObject(s3Bucket, IMAGE_JPX);
+            LOGGER.error(failure.getMessage());
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests clearing Cantaloupe's cache with null imageID
+     *
+     * @param aContext
+     */
+    @Test
+    public void testNoImageID(final TestContext aContext) {
+        final String imageIDNull = null;
+        final Async asyncTask = aContext.async();
+
+        deployNewVerticle(myConfigs).compose(verticleID -> {
+            myVertID = verticleID;
+            return sendMessage(imageIDNull);
+        }).onFailure(failure -> {
+            if (Integer.parseInt(failure.getMessage()) == HTTP.INTERNAL_SERVER_ERROR) {
+                TestUtils.complete(asyncTask);
+            } else {
+                aContext.fail();
+            }
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests that verticles fails when given null credentials
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    public void testNullCredential(final TestContext aContext) {
+        final Async asyncTask = aContext.async();
+        final JsonObject nullConfigs = myConfigs;
+
+        nullConfigs.put(IIIIF_USERNAME, (String) null);
+        nullConfigs.put(IIIIF_PASSWORD, (String) null);
+
+        deployNewVerticle(nullConfigs).onFailure(failure -> {
+            TestUtils.complete(asyncTask);
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Tests that verticles fails when given invalid credentials
+     *
+     * @param aContext A testing context
+     */
+    @Test
+    public void testInvalidCredential(final TestContext aContext) {
+        final Async asyncTask = aContext.async();
+        final JsonObject newConfigs = myConfigs;
+
+        newConfigs.put(IIIIF_USERNAME, "username");
+        newConfigs.put(IIIIF_PASSWORD, "password");
+
+        deployNewVerticle(newConfigs).onFailure(failure -> {
+            TestUtils.complete(asyncTask);
+        }).onSuccess(success -> {
+            aContext.fail();
+        });
+    }
+
+    /**
+     * Gets info.json from Cantaloupe of specific image
+     */
+    private Future<Integer> getInfoJsonObject() {
```

Comment:
This method is now simply returning the width value, so should be renamed and Javadoc updated.